lightning/src/ln/chan_utils.rs

```diff
@@ -656,6 +666,8 @@ pub struct ChannelTransactionParameters {
 	pub counterparty_parameters: Option<CounterpartyChannelTransactionParameters>,
 	/// The late-bound funding outpoint
 	pub funding_outpoint: Option<chain::transaction::OutPoint>,
+	/// Are anchors used for this channel.  Boolean is serialization backwards-compatible
+	pub opt_anchors: Option<()>
```

Comment:
this is for serialization backwards compatibility - so that if the boolean is missing it will be read in as false.  as recommended by @TheBlueMatt on a previous PR.

Reply:
I believe at the time we didn't have the `(default_value, false)` TLV deserialization type and only had `option`. I don't care about changing it, but that is likely a better option going forward.

Reply:
upgraded to boolean

Reply:
Oh, shit, I'm sorry, I mis-remembered all the details here. Right, so the new change will work correctly forward-compatibility-wise, but not backwards-. ie even with `(default_value)` we'll always write it out as an even number and cause old clients to refuse to read the new data. We could tweak `default_value` to also not write if the value is default, or have some new TLV type, but I'm not sure if it matters enough to change it here.

Reply:
Should I switch it back to `Option<()>`?

Reply:
Yea, sorry about that.

Reply:
no worries, done